pkg/yang/yang.go

```diff
@@ -552,13 +557,40 @@ func (s *AnyXML) NName() string         { return s.Name }
 func (s *AnyXML) Statement() *Statement { return s.Source }
 func (s *AnyXML) Exts() []*Statement    { return s.Extensions }
 
+// An AnyData is defined in: http://tools.ietf.org/html/rfc7950#section-7.10
+//
+// AnyData are only expected in YANG 1.1 modules (those with a
+// "yang-version 1.1;" statement in the module).
+type AnyData struct {
```

Comment:
I think enumerating it is fine, they may diverge in the future.

If you did want to do the type thing, I would make AnyData be the defined one and AnyXML be the derived one.  Too bad we cant just say:

type AnyXML struct {
    AnyData
}

And then just define Kind

But that would mess up parsing.